src/pinto_lab_to_nwb/into_the_void/interfaces/holographicstimulationdatainterface.py

```diff
@@ -0,0 +1,374 @@
+                    description="The stimulated ROIs",
+                    imaging_plane=imaging_plane_name,
+                )
+            ],
+        )
+        metadata["Ophys"].update(dict(ImageSegmentation=default_image_segmentation))
+
+        # add default optogenetic stimulus site metadata
+        default_site_metadata = dict(
+            name="site",
+            description="The targeted location of the holographic stimulation.",
+            device=device_name,
+        )
+        metadata["Ophys"].update(dict(OptogeneticStimulusSite=[default_site_metadata]))
+
+        return metadata
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: Optional[dict] = None,
+        stub_test: bool = False,
+    ) -> None:
+        metadata_copy = deepcopy(metadata)
+
+        plane_segmentation_name = self.plane_segmentation_name
+        plane_suffix = plane_segmentation_name.replace("PlaneSegmentation", "")
+        imaging_plane_name = "ImagingPlane" + plane_suffix
+
+        add_imaging_plane(nwbfile=nwbfile, metadata=metadata_copy, imaging_plane_name=imaging_plane_name)
+
+        device_name = "device"
+        device = nwbfile.devices[device_name]
+
+        # Add stimulus pattern to NWBFile
+        num_revolutions = int(self.sequence_mark_points_list[0]["SpiralRevolutions"])
+        inter_stimulus_interval = int(self.sequence_mark_points_list[0]["InterPointDelay"]) / 1000
+        spiral_params = self.sequence_mark_points_list[0]["points"][0]
+        diameter = float(spiral_params["SpiralWidth"])
+        height = float(spiral_params["SpiralHeight"])
+        spiral_scanning = SpiralScanning(
+            name="stimulus_pattern",
+            diameter=diameter,
+            height=height,  # spiral size in microns is 15
+            number_of_revolutions=num_revolutions,
+            description="The spiral scanning pattern used for holographic stimulation.",
+            duration=self.cycle_relative_durations[0],  # duration of each spiral
+            number_of_stimulus_presentation=len(self.sequence_mark_points_list),  # or 220 (22 * 10)
+            inter_stimulus_interval=inter_stimulus_interval,
+        )
+        nwbfile.add_lab_meta_data(spiral_scanning)
+
+        # Add optogenetic stimulus site to NWBFile
+        stimulus_site_metadata = metadata_copy["Ophys"]["OptogeneticStimulusSite"][0]
+        stimulus_site_metadata.update(device=device)
+        stim_site = PatternedOptogeneticStimulusSite(**stimulus_site_metadata)
+        nwbfile.add_ogen_site(stim_site)
+
+        # Add SLM device to NWBFile
+        spatial_light_modulator_metadata = metadata_copy["Ophys"]["OptogeneticDevice"]["SpatialLightModulator"]
+        spatial_light_modulator_name = spatial_light_modulator_metadata["name"]
+        if spatial_light_modulator_name in nwbfile.devices:
+            raise ValueError(f"'{spatial_light_modulator_name}' already added to the NWBFile.")
```

Comment:
From perspective of usability by the lab, any problem with downgrading this to a warning and fetching it from the file instead of creating a new one in the line below?